### PR TITLE
Fixed Eq[Double].

### PR DESCRIPTION
**Pull request by alexknvl**

### CODE REVIEW

**Comment by NeQuissimus** [comment]
We don't want `-0.0d` and `0.0d` to be equal?

**Comment by alexknvl** [comment]
They are clearly not equal, since they have different observable behavior:
```scala
def check(x: Double): Boolean = (1/x) > 0
println(check(+0.0), check(-0.0))
// (true,false)
```
They might be equivalent in a particular domain if you don't care about the distinction, but then you can just use a newtype that does not allow you to observe the difference.

File: base/shared/src/main/scala/scalaz/data/ilist.scala [hunk 1]
```diff
@@ -210,6 +210,9 @@ object IListModule {
 
     def !!(n: Int): Maybe[A] = self.index(n)
 
+    def cross[B](b: IList[B]): IList[(A, B)] =
+      self.flatMap(a => b.map(b => (a, b)))
```
Comment (edmundnoble) [inline base/shared/src/main/scala/scalaz/data/ilist.scala]: Wait, isn't this just `.tuple2`?

Comment (alexknvl) [reply base/shared/src/main/scala/scalaz/data/ilist.scala]: Yes, I think so, will fix.

Comment (alexknvl) [reply base/shared/src/main/scala/scalaz/data/ilist.scala]: Nvm, there is no tuple2 yet.

**Pull request merged**